src/main/scala/strawman/collection/ArrayOps.scala

```diff
@@ -19,6 +19,7 @@ class ArrayOps[A](val xs: Array[A])
   protected[this] def seq: Seq[A] = iterableFactory.fromIterable(coll)
 
   def length = xs.length
+  @throws[ArrayIndexOutOfBoundsException]
   def apply(i: Int) = xs.apply(i)
```

Comment:
There are a few reasons I added the annotation there.

- It is a more specific exception than that thrown by its `super`
- Though somewhat unlikely, the class could conceivably be inherited from (I've inherited from `Ordering.Ops` before)
- Even if not necessary for the implementer, it's helpful for a caller to be able to see that it throws an exception

Additionally, I don't feel that there's a significant cost incurred by adding the annotation, though you and others may disagree.